### DISCUSSION TITLE
v1.1.7 nvm use version still need admin ?

### DISCUSSION THREAD

**Discussion by nethawkChen** [coreybutler/nvm-windows#1022] [Support]
In this Microsoft article (https://learn.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-windows), there is the following description:

![nvm woring](https://github.com/coreybutler/nvm-windows/assets/74844278/c907a7a2-9d5b-4f49-8435-e6618a325ee7)

As I do not have administrator privileges on my development device, I opted to use version 1.1.7. However, I encountered difficulties with the nvm user version command. Therefore, does using version 1.1.7 indeed allow one to bypass the need for administrator privileges?

![nvm1 1 7](https://github.com/coreybutler/nvm-windows/assets/74844278/48e56869-2683-4d26-8d8e-a64a6235142f)

**Comment by coreybutler** [comment]
The article is outdated/wrong. NVM4W requires the `SeCreateSymbolicLinkPrivilege` to run `nvm use`. Admin privileges will grant this, but you can also get this permission by [enabling Developer Mode](https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development#:~:text=Enable%20Developer%20Mode%20in%20Windows%2010%201%20From,click%20Yes%20to%20accept%20the%20change.%20See%20More.). All versions of NVM4W dating back to the very first version have this requirement because it is a Windows requirement. Version 1.1.7 triggers a UAC prompt to get these permissions when needed. UAC prompting was removed in v1.1.8 at user request and re-integrated in 1.1.9 due to more user requests.

Use v1.1.11, which has UAC triggering. You have to have admin privileges to install an app on Windows, so I recommend installing NVM4W v1.1.11 at `C:\nvm-windows`. When the installer asks where Node.js should be installed (symlink), use `C:\nodejs`. This will configure NVM4W to use directories that don't require special permissions. However; you will still need developer mode enabled or admin privileges to run `nvm use` (or an admin must grant your account `SeCreateSymbolicLinkPrivilege` privileges) due to the Windows requirement.

**Comment by nethawkChen** [comment]
Is it possible to operate in Developer Mode without admin privileges? It seems that my testing indicates otherwise. I am on a Windows 10 environment, with Developer Mode enabled.

I am using the nvm-noinstall.zip method for a portable setup. I've created a settings.txt file in its directory with the following contents:
root: D:\Programs\nvm
path: D:\Programs\nodejs
arch: 64
proxy:

I have also added the following to my system environment variables:
NVM_HOME=D:\Programs\nvm
NVM_SYMLINK=D:\Programs\nodejs
PATH=%NVM_HOME%;%NVM_SYMLINK%

Afterward, I manually downloaded Node.js version 18.17.1, extracted it, placed it under D:\Programs\nvm, and renamed the folder to v18.17.1. Similarly, I downloaded Node.js version 16.20.2, repeated the same process, and renamed the folder to v16.20.2.

These operations work as expected under an account with local administrator privileges; nvm use version functions properly. However, for accounts without administrator privileges, even with the Developer Mode option enabled, the nvm use command remains unusable.

**Reply by coreybutler** [reply]
Yes, you can operate in Developer Mode without admin privileges. That should grant you `SeCreateSymbolicLinkPrivilege` privileges, but it doesn't necessarily grant you permissions to work with each directory.

If a manual download works but an `nvm install` doesn't work, then it may be a problem with permissions on the specific directories. I would compare the permissions between an `nvm install` version and a version you manually install for differences. Is your `D:` drive a shared directory or just a different hard drive? If it is a shared/mapped directory, make sure your account can symlink to the directory.

**Reply by nethawkChen** [reply]
D: drive is a different hard drive, not a shared or mapped drive.

**Reply by mklemarczyk** [reply]
@nethawkChen with your case in mind I posted the PR. In company that I work we encountered similar issue. As the system administrators are not willing to collaborate, there I present to you symlink_junction configuration option. 😃 

#1139
